### DISCUSSION TITLE
[thumbfast]subprocess failed:killed

### DISCUSSION THREAD

**Discussion by redomCL** [dyphire/mpv-config#106] [Q&A] [answered]
更新mpv和你的thumbfast后控制台有一条提示：[thumbfast]subprocess failed:killed，这个是不是就是https://github.com/dyphire/mpv-config/commit/c8ecdc55c294d4f473887d34e4284c22cacd7812 “thumbfast 脚本更新：在子进程中禁用上游新的媒体控制特性”？属于预期内得日志吗？感谢

**Comment by dyphire** [comment]
~~当然不是预期行为，无法复现~~

编辑：我在控制台没有看到此提示，打开 mpv 日志后看到了。这和引用的提交还有 mpv 更新都无关，使用之前的 thumbfast 脚本或旧版 mpv 一样可以在日志中看到它

应该是 thumbfast 脚本的小问题，不影响实际使用

更新：该 failed 仅在文件退出和切换时发生

**Reply by redomCL** [reply]
是的，我是在切换视频文件时，控制台触发的

**Reply by dyphire** [reply]
可以向 thumbfast 上游反馈，应该是某个事件监测导致的，不影响使用懒得排查（

**Reply by zhongfly** [reply]
https://github.com/mpv-player/mpv/pull/14524 应该是自从mpv增加子进程错误打印后出现的，之前脚本自己没有要求打印这个错误，现在mpv自己会打印这个错误

**Reply by zhongfly** [reply, marked as answer]
略微深入分析：
可以看mpv手册，其中子进程有一个选项感觉就和这个有关：https://mpv.io/manual/master/#command-interface-playback-only 默认是true，作用是当前播放结束时杀死进程，描述上看和这里提到的问题相符。
再看thumbfast的相关代码：https://github.com/po5/thumbfast/blob/f1fdf10b17f394f2d42520d0e9bf22feaa20a9f4/thumbfast.lua#L73
除了not pre_0_30_0且async为false时，会设置playback_only为false外，其他情况要么是true，要么没设置(那就会用默认值true)。而脚本实际上压根不存在playback_only为false的情况，因为它写死了pre_0_30_0是true

综上，应该完全是预期行为

**Reply by redomCL** [reply]
> 略微深入分析： 可以看mpv手册，其中子进程有一个选项感觉就和这个有关：https://mpv.io/manual/master/#command-interface-playback-only 默认是true，作用是当前播放结束时杀死进程，描述上看和这里提到的问题相符。 再看thumbfast的相关代码：https://github.com/po5/thumbfast/blob/f1fdf10b17f394f2d42520d0e9bf22feaa20a9f4/thumbfast.lua#L73 除了not pre_0_30_0且async为false时，会设置playback_only为false外，其他情况要么是true，要么没设置(那就会用默认值true)。而脚本实际上压根不存在playback_only为false的情况，因为它写死了pre_0_30_0是true
> 
> 综上，应该完全是预期行为

大佬好，综合上面打印的解释，能否理解为结束当前播放时正常的结束进程，但是以前控制台并不会输出这个操作的消息，现在输出这条消息了

**Reply by zhongfly** [reply]
是这个意思，之前脚本和mpv都不会打印这个的消息，现在mpv会打印了。一直都有这样的操作
